### ISSUE TITLE
False Positive When Using SeparateDatabaseAndState

### ISSUE THREAD

**Issue by johnnymetz** [3YOURMIND/django-migration-linter#294]
## Steps to Reproduce

Let's say I have the following Django model:

```
class Product(models.Model):
    name = models.CharField(max_length=255)
    rating = models.IntegerField(null=True, blank=True)
```

I want to remove the rating field in a backwards-compatible way, so I create the following two migrations:

**Migration 1: Remove rating field from the state**

```
# 0002_remove_product_rating_from_state.py
class Migration(migrations.Migration):
    dependencies = [
        ("appname", "0001_initial"),
    ]

    operations = [
        migrations.SeparateDatabaseAndState(
            state_operations=[
                migrations.RemoveField(
                    model_name="product",
                    name="rating",
                ),
            ],
            database_operations=[],
        ),
    ]
```

**Migration 2: Remove rating field from the database**

```
# 0003_remove_product_rating_from_db.py
class Migration(migrations.Migration):
    dependencies = [
        ("appname", "0002_remove_product_rating_from_state"),
    ]

    operations = [
        migrations.SeparateDatabaseAndState(
            state_operations=[],
            database_operations=[
                migrations.RunSQL(
                    "ALTER TABLE product DROP COLUMN rating;"
                ),
            ],
        ),
    ]
```

The `lintmigrations` command reports that `0003_remove_product_rating_from_db` is backwards incompatible:

```
$ python manage.py lintmigrations -q ok
(api, 0004_remove_metabase_field_from_db)... ERR
	DROPPING columns
```

## Expected Result

The `lintmigrations` should show no errors. The `rating` field was removed from the state in `0002_remove_product_rating_from_state` which makes `0003_remove_product_rating_from_db` backwards compatible.

## Questions / Comments

Does this seem like the correct solution? One issue is that these two migrations are ONLY backwards compatible if they're deployed separately. Deploying them both at once is obviously backwards incompatible.

If we can't track / enforce when these migrations are deployed, is the recommended solution to continue to report as backwards incompatible and require the user to ignore the false positive?

```
$ python manage.py lintmigrations -q ok -- ignore_name 0004_remove_metabase_field_from_db
# No errors
```

**Comment by lukas-vokracko-b2c2** [comment]
+1

**Comment by David-Wobrock** [comment]
Hey @johnnymetz,

Thanks for opening an issue about this.

`SeparateDatabaseAndState` are not really properly handled by the django-migration-linter, but it would be a good addition 👍 

> One issue is that these two migrations are ONLY backwards compatible if they're deployed separately. Deploying them both at once is obviously backwards incompatible.

I reckon you are correct about this.
I think a user splitting the migration into two using `SeparateDatabaseAndState` has a relatively advanced usage of migrations, and their intent is to deploy them separately.
On top of my head, I would suggest the django-migration-linter raises a warning in this case, highlighting to be careful to really deploy it in two different steps. Else, raising an error would be fine as well IMO, as the tool should be rather pessimistic.
